components/Tabs/tests/snapshot/MDCTabBarExtendedAlignmentSnapshotTests.m

```diff
@@ -0,0 +1,293 @@
+  self.item3.badgeValue = kItemTitleShort1Arabic;
+  self.item4.title = kItemTitleShort1Arabic;
+  self.item5.title = kItemTitleShort2Arabic;
+}
+
+- (void)setTitlesToArabicLong {
+  self.item1.title = kItemTitleLong1Arabic;
+  self.item2.title = kItemTitleLong2Arabic;
+  self.item3.title = kItemTitleLong3Arabic;
+  self.item3.badgeValue = kItemTitleLong1Arabic;
+  self.item4.title = kItemTitleLong1Arabic;
+  self.item5.title = kItemTitleLong2Arabic;
+}
+
+- (void)generateSnapshotAndVerifyForView:(UIView *)view {
+  UIView *snapshotView = [view mdc_addToBackgroundView];
+  [self snapshotVerifyView:snapshotView];
+}
+
+#pragma mark - Alignment tests
+
+- (void)testTabBarTitlesOnlyFitLatinLTR {
+  // When
+  self.tabBar.frame = CGRectMake(0, 0, 480, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitlesOnlyFitArabicRTL {
+  // When
+  [self setTitlesToArabicShort];
+  [self changeLayoutToRTL];
+  self.tabBar.frame = CGRectMake(0, 0, 480, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitlesOnlyDontFitLatinLTR {
+  // When
+  [self setTitlesToLatinLong];
+  self.tabBar.frame = CGRectMake(0, 0, 480, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitlesOnlyDontFitArabicRTL {
+  // When
+  [self setTitlesToArabicLong];
+  [self changeLayoutToRTL];
+  self.tabBar.frame = CGRectMake(0, 0, 480, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitledImagesAppearanceFitLatinLTR {
+  // When
+  self.tabBar.itemAppearance = MDCTabBarItemAppearanceTitledImages;
+  self.tabBar.frame = CGRectMake(0, 0, 480, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitledImagesAppearanceFitArabicRTL {
+  // When
+  self.tabBar.itemAppearance = MDCTabBarItemAppearanceTitledImages;
+  [self setTitlesToArabicShort];
+  [self changeLayoutToRTL];
+  self.tabBar.frame = CGRectMake(0, 0, 480, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitledImagesAppearanceDontFitLatinLTR {
+  // When
+  self.tabBar.itemAppearance = MDCTabBarItemAppearanceTitledImages;
+  self.tabBar.frame = CGRectMake(0, 0, 200, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarTitledImagesAppearanceDontFitArabicRTL {
+  // When
+  self.tabBar.itemAppearance = MDCTabBarItemAppearanceTitledImages;
+  [self setTitlesToArabicShort];
+  [self changeLayoutToRTL];
+  self.tabBar.frame = CGRectMake(0, 0, 200, 100);
+  [self.tabBar sizeToFit];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.tabBar];
+}
+
+- (void)testTabBarImagesAppearanceFitLatinLTR {
```

Comment:
The screenshot for testTabBarImagesAppearanceFitLatinLTR_11_2@2x.png seems off to me. Shouldn't it be justified evenly across the screen? The testTabBarImagesAppearanceFitArabicRTL_11_2@2x.png, for instance, looks like what I'd expect.

Reply:
Yeah, it's annoying. It's https://github.com/material-components/material-components-ios/issues/7116 rearing its ugly head again. The fix hasn't been prioritized, though.

Reply:
Should we prioritize that issue in relation to this work?

Reply:
Tabs shouldn't have badges, but some clients have asked us to have it. I would think that fixing the layout is worthwhile, but it's not clear we should fix it since we don't technically support the badging in the first place.

Reply:
@mrhappyasthma will this behavior negatively affect y'all?

Reply:
We don't currently use badges on our tabs, so this won't be an immediate issue. However, historically, I find PMs tend to love using badging on various UI elements to draw attention to a new feature, so I wouldn't be surprised if someone eventually wanted this behavior.

So probably something worth fixing but not a blocker for my team.